src/Root.tsx

```diff
@@ -48,6 +52,48 @@ export async function onLatestVersionChange(newVer: SemVer | null | undefined, o
 const Root = observer(() => {
   const { app, modals } = getRootStore();
 
+  React.useEffect(() => {
+    const searchParams = new URLSearchParams(document.location.search);
+    const result = searchParams.get("diagram");
+    if (result === null) return;
+
+    // Replace URL-safe characters with base64 equivalents
+    let base64String = result.replaceAll("-", "+").replaceAll("_", "/");
+
+    try {
+      const diagramDataInJson = window.atob(base64String);
+      const c = plainToClass(Diagram, JSON.parse(diagramDataInJson), {
+        excludeExtraneousValues: true,
+        exposeDefaultValues: true
+      });
+      validate(c).then(errors => {
+        if (errors.length > 0) {
+          const data1Buttons: ConfirmationButton[] = [{ label: "OK" }];
```

Comment:
I don't think you need to declare `data1Buttons` and `data1`, you can put them in the same line as the `confirmation.prompt`.

Reply:
Updated, thank you